lib/spack/spack/binary_distribution.py

```diff
@@ -137,15 +145,21 @@ def read_buildinfo_file(prefix):
     return buildinfo
 
 
-def write_buildinfo_file(prefix, workdir, rel=False):
+def write_buildinfo_file(spec, workdir, rel=False):
     """
     Create a cache file containing information
     required for the relocation
     """
+    prefix = spec.prefix
     text_to_relocate = []
     binary_to_relocate = []
     link_to_relocate = []
     blacklist = (".spack", "man")
+    prefix2hash = dict()
+    prefix2hash[str(spec.package.prefix)] = spec.format('{hash}')
```

Comment:
The hash that `spec.format('{hash}')` resolves to might change, so could this be `spec.dag_hash()` for example (i.e. choose whatever hash this would resolve to now).